connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -231,4 +248,182 @@ public void transitionTo(TargetState state) {
+//            Percentiles commitPercentiles = new Percentiles(4 * buckets,
+//                                                            0.0,
+//                                                            10*1000.0,
+//                                                            BucketSizing.LINEAR,
+//                                                            new Percentile(p50, 50),
+//                                                            new Percentile(p75, 75),
+//                                                            new Percentile(p90, 90),
+//                                                            new Percentile(p95, 95),
+//                                                            new Percentile(p99, 99));
+//            commitTime.add(commitPercentiles);
+
+            batchSize = metricGroup.metrics().sensor("batch-size");
+            batchSize.add(metricGroup.metricName("batch-size-max",
+                    "The maximum size of the batches processed by the connector"),
+                    new Max());
+            batchSize.add(metricGroup.metricName("batch-size-avg",
+                    "The average size of the batches processed by the connector"),
+                    new Avg());
+
+            MetricName offsetCommitFailures = metricGroup.metricName("offset-commit-failure-percentage",
+                    "The average percentage of this task's offset commit attempts that failed");
+            MetricName offsetCommitSucceeds = metricGroup.metricName("offset-commit-success-percentage",
+                    "The average percentage of this task's offset commit attempts that failed");
+            Frequencies commitFrequencies = Frequencies.forBooleanValues(offsetCommitFailures, offsetCommitSucceeds);
+            commitAttempts = metricGroup.metrics().sensor("offset-commit-completion");
+            commitAttempts.add(commitFrequencies);
+        }
+
+        private void addTaskStateMetric(final State matchingState, String name, String description) {
+            metricGroup.addIndicatorMetric(matchingState, name, description, new Supplier<State>() {
+                @Override
+                public State get() {
+                    return taskStateTimer.currentState();
+                }
+            });
+        }
+
+        private void addRatioMetric(final State matchingState, String name, String description) {
+            MetricName metricName = metricGroup.metricName(name, description);
+            if (metricGroup.metrics().metric(metricName) == null) {
+                metricGroup.metrics().addMetric(metricName, new Measurable() {
+                    @Override
+                    public double measure(MetricConfig config, long now) {
+                        return taskStateTimer.durationRatio(matchingState, now);
+                    }
+                });
+            }
+        }
+
+        void recordCommit(long duration, boolean success, Throwable error) {
```

Comment:
`error` is unused

Reply:
I anticipate adding more metrics in the future that may want to know the error (e.g., last error reported by the task). I'd rather leave this as is.